src/transcription/prerecorded/audio_source.rs

```diff
@@ -2,67 +2,69 @@ use reqwest::{header::CONTENT_TYPE, RequestBuilder};
 use serde::Serialize;
 
 /// Used as a parameter for [`Transcription::prerecorded`](crate::transcription::Transcription::prerecorded) and similar functions.
-///
-/// This trait is [sealed], and thus cannot be implemented by anything new.
-///
-/// [sealed]: https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed
-pub trait AudioSource: private::Sealed {
-    #[doc(hidden)]
-    fn fill_body(self, request_builder: RequestBuilder) -> RequestBuilder;
-}
-
-/// Used to prevent other crates from implementing AudioSource
-/// See <https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed>
-mod private {
-    use super::{BufferSource, UrlSource};
+#[derive(Debug)]
+pub struct AudioSource(InternalAudioSource);
 
-    pub trait Sealed {}
-
-    impl Sealed for &UrlSource<'_> {}
-    impl<B: Into<reqwest::Body>> Sealed for BufferSource<'_, B> {}
+#[derive(Debug)]
+enum InternalAudioSource {
+    Url(String),
```

Comment:
Similarly, this could be the [`url::Url`](https://docs.rs/url/latest/url/struct.Url.html) type, though I'm less sold on that, since it doesn't remove the need for creating a string in any case.